File: 3634-find-mirror-score-of-a-string/find-mirror-score-of-a-string.cpp
class Solution {
public:
    long long calculateScore(string s) {
        long long ans = 0;
        vector<stack<int>> stk(26);

        for (int i = 0; i < s.size(); ++i) {
            int v = s[i] - 'a';
            if (stk[25 - v].size()) {
                ans += i - stk[25 - v].top();
                stk[25 - v].pop();
            } else {
                stk[v].push(i);
            }
        }

        return ans;
    }
};

File: 3634-find-mirror-score-of-a-string/find-mirror-score-of-a-string.py
class Solution:
    def calculateScore(self, s: str) -> int:
        ans = 0
        stacks = [[] for _ in range(26)]

        for i, ch in enumerate(s):
            v = ord(ch) - 97
            if stacks[25 - v]:
                ans += i - stacks[25 - v].pop()
            else:
                stacks[v].append(i)
                
        return ans
